pkg/k8s/client/fake.go

```diff
@@ -37,21 +37,26 @@ import (
 	"github.com/cilium/cilium/pkg/k8s/testutils"
 )
 
-var FakeClientCell = cell.Module(
-	"k8s-fake-client",
-	"Fake Kubernetes client",
-
-	cell.ProvidePrivate(
-		newStateDBObjectTracker,
-	),
-
-	cell.Provide(
-		NewFakeClientsetWithTracker,
-		func(fc *FakeClientset) hive.ScriptCmdsOut {
-			return hive.NewScriptCmds(FakeClientCommands(fc))
-		},
-	),
-)
+// We do not create a cell as global variable since
+// compiler will compile all of the fake protobufs into the release
+// binary which increases the binary size by ~20 MB
+var FakeClientCell = func() cell.Cell {
```

Comment:
It this change actually necessary after moving the fake client to the dedicated test package?